doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1506,160 +1494,140 @@ Managing the local context
-   This lets you avoid specifying the asserted statement completely before starting
-   to prove it.
+   .. exn:: The term "@type" has type "@type__1" which should be Set, Prop or Type.
 
-.. tacv:: pose proof @term {? as @simple_intropattern}
-   :name: pose proof
+      Arises when the argument :token:`type` is neither of type :g:`Prop`,
+      :g:`Set` nor :g:`Type`.
 
-   This tactic behaves like :n:`assert @type {? as @simple_intropattern} by exact @term`
-   where :token:`type` is the type of :token:`term`. In particular,
-   :n:`pose proof @term as @ident` behaves as :n:`assert (@ident := @term)`
-   and :n:`pose proof @term as @simple_intropattern` is the same as applying the
-   :token:`simple_intropattern` to :token:`term`.
+   .. exn:: Proof is not complete.
+      :name: Proof is not complete. (assert)
 
-.. tacv:: epose proof @term {? as @simple_intropattern}
-   :name: epose proof
+      :n:`@ltac_expr3` was not able to prove the new hypothesis.
 
-   While :tacn:`pose proof` expects that no existential variables are generated by
-   the tactic, :tacn:`epose proof` removes this constraint.
+   .. tacn:: eassert ( @ident : @type ) {? by @ltac_expr3 }
+             eassert ( @ident := @term )
+             eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eassert; _; _
 
-.. tacv:: pose proof (@ident := @term)
+      Unlike :tacn:`assert`, the :n:`@type`\s, :n:`@term`\s and :n:`@one_term`\s in
+      :tacn:`eassert` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-   This is an alternative syntax for :n:`assert (@ident := @term)` and
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
+.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
+          enough @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: enough; _
 
-.. tacv:: epose proof (@ident := @term)
+   .. todo PR The only difference vs assert is where the goal is added?  If so, then
+      this entry should only refer to assert and mention the difference(s).
+      There are several similar tactics in this section.  Maybe we should add a
+      summary of their differences?  Perhaps you could briefly outline them?
 
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+   The first form adds a new hypothesis :n:`@ident : @type` to the context
+   and adds :n:`@type` as a new subgoal after the current goal (rather than
+   before it as :tacn:`assert` would).  If present, :n:`@ltac_expr3` must solve the
+   initial goal entirely after the extra assumption :n:`@type` is added and possibly
+   destructed; otherwise, the tactic will fail.
```

Comment:
If this first paragraph remains solely about the first form, then there is no possible destructuring of the extra assumption:
```suggestion
   initial goal entirely after the extra assumption :n:`@type` is added; otherwise, the tactic will fail.
```

Reply:
Ah, destructing is not relevant to the tactic at all?

Reply:
Destructuring can only happen if there is an intropattern (so the second form).

Reply:
"destructuring" and not "destructing"?  If it's something the intropattern does, then no need to use one of those terms here.  RIght?

Reply:
>"destructuring" and not "destructing"?

Might be a Frenchism again. In French, we have "destructurer" and "détruire", but they probably in fact correspond to "destruct" and "destroy". So "destructing" is probably the right term.

> If it's something the intropattern does, then no need to use one of those terms here. RIght?

Right.